File: backend/paperchat_backend/models/bootstrap.py
from enum import StrEnum
from typing import Literal

from pydantic import BaseModel


class HealthResponse(BaseModel):
    status: Literal["ok"]


class CheckResult(BaseModel):
    ok: bool
    message: str | None = None


class BootstrapError(BaseModel):
    code: str
    message: str
    action: str


class BootstrapStatus(StrEnum):
    ready = "ready"
    starting = "starting"
    degraded = "degraded"
    error = "error"


class BootstrapResponse(BaseModel):
    app_version: str
    status: BootstrapStatus
    checks: dict[str, CheckResult]
    errors: list[BootstrapError]

File: frontend/src/api/bootstrap.ts
import { fetchJson } from "./client";

export interface CheckResult {
  ok: boolean;
  message: string | null;
}

export interface BootstrapError {
  code: string;
  message: string;
  action: string;
}

export type BootstrapStatus = "ready" | "starting" | "degraded" | "error";

export interface BootstrapResponse {
  app_version: string;
  status: BootstrapStatus;
  checks: Record<string, CheckResult>;
  errors: BootstrapError[];
}

export interface HealthResponse {
  status: string;
}

export function fetchHealth(): Promise<HealthResponse> {
  return fetchJson<HealthResponse>("/api/health");
}

export function fetchBootstrap(): Promise<BootstrapResponse> {
  return fetchJson<BootstrapResponse>("/api/bootstrap");
}
